src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,106 @@
+  const [todosList, setTodosList] = useState<Todo[]>([]);
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+  const { query, status } = useAppSelector(state => state.filter);
 
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">laboriosam mollitia et enim quasi adipisci quia provident illum</p>
-            </td>
+  const handleClickSelect = (todo: Todo) => dispatch(currentTodoActions.setTodo(todo));
 
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">6</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">qui ullam ratione quibusdam voluptatem quia omnis</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">7</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">illo expedita consequatur quia in</p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
+  useEffect(() => {
+    const filteredTodos = todos
+      .filter(todo => todo.title.toLocaleLowerCase().includes(query.toLocaleLowerCase()))
+      .filter(todo => {
+        switch (status) {
+          case 'completed':
+            return todo.completed;
 
-          <tr data-cy="todo">
-            <td className="is-vcentered">8</td>
+          case 'active':
+            return !todo.completed;
 
-            <td className="is-vcentered">
-              <span className="icon" data-cy="iconCompleted">
-                <i className="fas fa-check" />
-              </span>
-            </td>
+          default:
+            return true;
+        }
+      });
 
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-success">quo adipisci enim quam ut ab</p>
-            </td>
+    setTodosList(filteredTodos);
```

Comment:
you don't need `todosList` state - you calculate it based on other state/redux saved values. Just create `filteredTodos` variable, using `useMemo` hook with the same list of dependencies